lib/ex_json_path.ex

```diff
@@ -17,14 +17,16 @@
 #
 
 defmodule ExJsonPath do
-  def eval(map, path) when is_map(map) and is_binary(path) do
+  defguard is_valid_input(input) when is_map(input) or is_list(input)
```

Comment:
I think I recall that also bare JSON values are valid JSON (e.g. `3` is a valid JSON document).
In fact, if you put [here](https://jsonpath.com/) `"foo"` as JSON and `$` as JSON path query, it will return `["foo"]` as result.

I think this can be handled in a special function head since the only possible query on raw values is `$`, but I might be wrong.